### PR TITLE
unbound 1.5.7 and fix conf file tweaking

### PR DESCRIPTION
**Pull request by apjanke**
Follows up on #46858

### CODE REVIEW

File: Library/Formula/unbound.rb [hunk 1]
```diff
@@ -19,13 +19,10 @@ def install
                           "--sysconfdir=#{etc}",
                           "--with-libevent=#{Formula["libevent"].opt_prefix}",
                           "--with-ssl=#{Formula["openssl"].opt_prefix}"
+    inreplace "doc/example.conf", 'username: "unbound"', "username: \"#{ENV["USER"]}\""
```
Comment (mikemcquaid) [inline Library/Formula/unbound.rb]: This won't work unfortunately because it needs to use the user at `post_install` rather than bottling time.

Comment (DomT4) [inline Library/Formula/unbound.rb]: Aye. You could always do something like:

```ruby
def post_install
  return if File.read(etc/"unbound/unbound.conf").include?("#{ENV["USER"]}")
  inreplace etc/"unbound/unbound.conf", 'username: "unbound"', "username: \"#{ENV["USER"]}\""
end
```

Comment (mikemcquaid) [inline Library/Formula/unbound.rb]: Or could just make the `inreplace` essentially unconditional: `inreplace etc/"unbound/unbound.conf", 'username: ".*"', "username: \"#{ENV["USER"]}\""`

Comment (apjanke) [inline Library/Formula/unbound.rb]: I think the `inreplace` needs to remain conditional, on the principle that if the user has changed the config file, we shouldn't go changing it further? A `brew install` run should only change that file if it was placed there by that same run. Same reason we don't overwrite existing files in `etc`.

Anyway, /me smacks forehead for not realizing why it won't work.

Comment (apjanke) [inline Library/Formula/unbound.rb]: How about something like this: we do a build-time replacement with a sentinel value, so we can tell if it's a "default" config file that this installation run just dropped, and then conditionally replace it.

```
def install
  ...
    inreplace "doc/example.conf", 'username: "unbound"', 'username: "default-unbound-user"'
  ...
end

def post_install
  if File.read(etc/"unbound/unbound.conf").include?('username: "default-unbound-user"')
    inreplace etc/"unbound/unbound.conf", 'username: "default-unbound-user"', "username: \"#{ENV["USER"]}\""
  end
end

Or maybe the test should be to check whether that `etc/unbound/unbound.conf` file exists at `pre_install` time, and if so, then do the replacement in `post_install`. That seems more congruent with the "only modify files that this installation run has created" intent.

Comment (mikemcquaid) [inline Library/Formula/unbound.rb]: Seems fine to me but maybe use something like `@@HOMEBREW-UNBOUND-USER@@` as the placeholder.

Comment (apjanke) [inline Library/Formula/unbound.rb]: I've revised the commit to do so.

Works over multiple reinstalls when I'm testing locally. And it should play nice with installs from the previous versions of the formula: they won't contain the sentinel value in their config files, so they will be silently left alone, which is what we want.

I'll try testing it with the bottle once the test bot run finishes.

**Comment by apjanke** [comment]
Manual testing of the bottle on OS X 10.9 looks good. Config file is left alone, and there are no errors or warnings on reinstalls. Merged.

**Pull request closed without merging**